### PR TITLE
[Docs] Predefined pages

### PR DESCRIPTION
**Pull request by johnnyomair**
## Description

Describes what is necessary to support predefined pages in a Comet Project.

## Related tasks and documents

Deploy preview: https://deploy-preview-2575.docs.comet-dxp.com/docs/documents/predefined-pages

Task: https://vivid-planet.atlassian.net/browse/COM-1080

## Open TODOs/questions

- [ ] Merge https://github.com/vivid-planet/comet/pull/2550
- [x] Merge https://github.com/vivid-planet/comet/pull/2572

## Further information

I haven't documented how to to get structured data (e.g., news) to work in the Site with Next App Router. IMO this should be a separate guide.

### CODE REVIEW

File: docs/docs/documents/predefined-pages.mdx [hunk 1]
```diff
@@ -0,0 +1,270 @@
...
+
+#### Adding the predefined pages to the middleware
+
+We use a combination of a redirect and a rewrite in the middleware to ensure that a predefined page can be viewed at the expected location in the page tree.
+
+**Example:** The news page in the code at `https://example.com/news` should be rendered in the page tree at `https://example.com/company/news`.
+We need the following redirect and rewrite:
+
+-   A **rewrite** from `/company/news` to `/news` (to ensure that the page can be viewed at the expected location)
+-   A **redirect** from `/news` to `/company/news` (to avoid duplicate content)
```
Comment (nsams) [inline docs/docs/documents/predefined-pages.mdx:179]: I would rather use translation of the url as example, as adding a fixed /company in the url should be done using static routes instead

File: docs/docs/documents/predefined-pages.mdx [hunk 2]
```diff
@@ -0,0 +1,270 @@
...
+## Adding a new predefined page type
+
+If you want to add a new type (e.g., products), the following changes must be made:
+
+1. API: Add the new type to the GraphQL enum:
+
+    ```diff title="src/documents/predefined-pages/entities/predefined-page.entity.ts"
+    export enum PredefinedPageType {
+        News = "News",
+    +   Products = "Products",
```
Comment (nsams) [inline docs/docs/documents/predefined-pages.mdx:250]: in the introduction note you write "This guide uses news as an example of structured data", you should also use news here, I suppose

Comment (johnnyomair) [reply docs/docs/documents/predefined-pages.mdx:250]: This section is intended to show what's necessary to add another predefined page type, so I used products as a second example. Isn't this clear enough? I could switch products and news here though.

Comment (thomasdax98) [reply docs/docs/documents/predefined-pages.mdx:250]: I think it's okay. At least it was clear to me

File: docs/docs/documents/predefined-pages.mdx (hunk 1 above)
[the remark this answers is not in the document]
Comment (johnnyomair) [reply docs/docs/documents/predefined-pages.mdx:179]: Yes. I'll try to rewrite this for more clarity

File: docs/docs/documents/predefined-pages.mdx [hunk 3]
```diff
@@ -0,0 +1,270 @@
...
+1.  Copy the predefined pages [helper functions](https://github.com/vivid-planet/comet/tree/main/demo/site/src/middleware/predefinedPages.ts) from the Demo project
+
+    :::warning
+
+    Support for filtering by document type in the `paginatedPageTreeNodes` query has been added in version 7.4.0.
+    Make sure your project uses Comet 7.4.0 or later.
+
+    :::
+
+2.  Add the rewrite and redirect to the middleware:
```
Comment (johnnyomair) [inline docs/docs/documents/predefined-pages.mdx:192]: I'm not sure if this works for Comet projects with site configs. I'll test that once we've added site configs to Demo.

File: docs/docs/documents/predefined-pages.mdx (hunk 1 above)
Comment (johnnyomair) [reply docs/docs/documents/predefined-pages.mdx]: ecd2e4880e7d2c193d62f16ef1ce4585d637d867

[the remark this answers is not in the document]
Comment (johnnyomair) [reply docs/docs/documents/predefined-pages.mdx]: ecd2e4880e7d2c193d62f16ef1ce4585d637d867

File: docs/docs/documents/predefined-pages.mdx (hunk 2 above)
Comment (johnnyomair) [reply docs/docs/documents/predefined-pages.mdx:252]: 984e8193c46cf0862ed1fa57dff0504b919b4567

**Review by thomasdax98 — APPROVED** [review]

**Review by nsams — APPROVED** [review]

**Pull request merged**